### PR TITLE
Refactor: extract creation of basic claim from issuance relative to prev state

### PR DESCRIPTION
**Pull request by nedgar**
I'm experimenting with different types of claims, e.g. for document verification. 
The changes here help isolate the basic claim creation from the rest of the issuance logic:
- extract the basic claim creation into its own `createBasicClaim` method
- extract the rest of the issuance into an `issueClaim` method, separate from the subcommand args processing and validation in `IssueClaim`
- rename `ageClaim` to `basicClaim` etc.
- a simplification of the hashing

See separate commits for these, but feel free to squash.

FYI @jimthematrix @Chengxuan

### CODE REVIEW

**Comment by nedgar** [comment]
Also changed to pass the IDs and nonce by value.

**Comment by nedgar** [comment]
Rebased on latest main branch.

**Review by Chengxuan — APPROVED** [review]

File: identity/internal/claim.go [hunk 1]
```diff
@@ -73,12 +72,40 @@ func IssueClaim() {
...
+	// issue the claim relative to prev state and persist it
+	issueClaim(basicClaim, issuerNameStr, *issuerId, holderId, privKey, *revNonce)
+}
+
+func createBasicClaim(holderId core.ID, revNonce uint64) *core.Claim {
+	// Load the schema for the claim
+	schemaBytes, _ := os.ReadFile("./schemas/kyc.json-ld")
+	var sHash core.SchemaHash
+
+	h := keccak256.Hash(schemaBytes, []byte("AgeCredential"))
```
Comment (Chengxuan) [inline identity/internal/claim.go:89]: good catch on the hashing method for schema

**Review by Chengxuan — APPROVED** [review]
LGTM

**Pull request merged**